site/topics/programming/assembler.py

```diff
@@ -0,0 +1,79 @@
+import sys
+
+OPERATORS = {
+    "NOOP": 0b0000,
+    "LDAR": 0b0001,
+    "LDAD": 0b0010,
+    "LDBR": 0b0011,
+    "LDBD": 0b0100,
+    "SAVA": 0b0101,
+    "SAVB": 0b0110,
+    "ADDB": 0b0111,
+    "SUBB": 0b1000,
+    "JMPA": 0b1001,
+    "OUTU": 0b1101,
+    "OUTS": 0b1110,
+    "HALT": 0b1111,
+}
+HAS_OPERAND = {
+    "LDAR",
+    "LDAD",
+    "LDBR",
+    "LDBD",
+    "SAVA",
+    "SAVB",
+    "JMPA",
+    "OUTU",
+    "OUTS",
+}
+
+def parse_number(number_string:str) -> int:
+    try:
+        if number_string[:2] == "0b":
+            number = int(number_string, 2)
+        elif number_string[:2] == "0x":
+            number = int(number_string, 16)
+        else:
+            number = int(number_string)
+    except ValueError:
+        raise ValueError(f"Cannot parse operand {number_string}")
+    return number
```

Comment:
If you're not concerned with security, you could use, `int(eval(n))`.

Reply:
Works nice with the syntex regex.